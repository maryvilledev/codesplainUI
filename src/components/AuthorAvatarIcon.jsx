import React, { PropTypes } from 'react';
import {
  Avatar,
} from 'material-ui';
import ReactTooltip from 'react-tooltip';

const styles = {
  avatar: {
    flexBasis: 'auto',
    margin: '3px 3px 3px 20px',
    paddingTop: '10px',
    verticalAlign: 'bottom',
  },
};

const AuthorAvatarIcon = ({ author, avatarUrl }) => {
  if (avatarUrl) {
    return (
      <div
        data-for="author-avatar"
        data-tip
        style={styles.avatar}
      >
        <Avatar
          size={20}
          src={avatarUrl}
        />
        <ReactTooltip
          id="avatar"
          effect="solid"
          place="bottom"
        >
          {author}
        </ReactTooltip>
      </div>
    );
  }
  return null;
};

AuthorAvatarIcon.propTypes = {
  author: PropTypes.string,
  avatarUrl: PropTypes.string,
};

AuthorAvatarIcon.defaultProps = {
  author: '',
  avatarUrl: '',
};

export default AuthorAvatarIcon;
