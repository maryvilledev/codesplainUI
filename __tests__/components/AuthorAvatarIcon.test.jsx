import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import AuthorAvatarIcon from '../../src/components/AuthorAvatarIcon';

describe('<AuthorAvatarIcon />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });
  describe('snapshots', () => {
    it('matches snapshot when avatar url is provided', () => {
      const wrapper = shallowWithContext(
        <AuthorAvatarIcon
          avatarUrl="https://avatars2.githubusercontent.com/u/0?v=3"
          author="author"
        />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('matches snapshot when no avatar url is provided', () => {
      const wrapper = shallowWithContext(
        <AuthorAvatarIcon
          avatarUrl=""
          author=""
        />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
