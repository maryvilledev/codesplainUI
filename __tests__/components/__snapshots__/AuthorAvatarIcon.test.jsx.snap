exports[`<AuthorAvatarIcon /> snapshots matches snapshot when avatar url is provided 1`] = `
<div
  data-for="author-avatar"
  data-tip={true}
  style={
    Object {
      "flexBasis": "auto",
      "margin": "3px 3px 3px 20px",
      "paddingTop": "10px",
      "verticalAlign": "bottom",
    }
  }>
  <Avatar
    size={20}
    src="https://avatars2.githubusercontent.com/u/0?v=3" />
  <ReactTooltip
    effect="solid"
    id="avatar"
    insecure={true}
    place="bottom"
    resizeHide={true}
    wrapper="div">
    author
  </ReactTooltip>
</div>
`;

exports[`<AuthorAvatarIcon /> snapshots matches snapshot when no avatar url is provided 1`] = `null`;
